Mathlib/RingTheory/RootsOfUnity/More.lean

```diff
@@ -0,0 +1,79 @@
+/-
+Copyright (c) 2024 Michael Stoll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Stoll
+-/
+import Mathlib.FieldTheory.KummerExtension
+import Mathlib.RingTheory.RootsOfUnity.Basic
+
+/-!
+# More results on primitive roots of unity
+
+(We put these in a separate file because of the `KummerExtension` import.)
+
+Assume that `μ` is a primitive `n`th root of unity in an integral domain `R`. Then
+$$ \prod_{k=1}^{n-1} (1 - \mu^k) = n \,; $$
+see `IsPrimitiveRoot.prod_one_sub_pow_eq_order` and its variant
+`IsPrimitiveRoot.prod_pow_sub_one_eq_order` in terms of `∏ (μ^k - 1)`.
+
+We use this to deduce that `n` is divisible by `(μ - 1)^k` in `ℤ[μ] ⊆ R` when `k < n`.
+-/
+
+/-- If `A` is an `R`-algebra and `μ : A`, then `μ^k-1` is divisible by `μ-1` in `R[μ] ⊆ A`. -/
+lemma Algebra.adjoin.sub_one_dvd_pow_sub_one (R : Type*) {A : Type*} [CommRing R] [CommRing A]
+    [Algebra R A] (μ : A) (k : ℕ) :
+    ∃ z ∈ adjoin R {μ}, μ ^ k - 1 = z * (μ - 1) := by
+  refine ⟨(Finset.range k).sum (μ ^ ·), ?_, (geom_sum_mul μ k).symm⟩
+  exact Subalgebra.sum_mem _ fun m _ ↦ Subalgebra.pow_mem _ (self_mem_adjoin_singleton _ μ) _
+
+variable {R : Type*} [CommRing R] [IsDomain R]
+
+namespace IsPrimitiveRoot
+
+open Finset Polynomial BigOperators
+
+/-- If `μ` is a primitive `n`th root of unity in `R`, then `∏(1≤k<n) (1-μ^k) = n`.
+(Stated with `n+1` in place of `n` to avoid the condition `n ≠ 0`.) -/
```

Comment:
I am not sure using `n+1` is a good idea here. I can be better in principle, but looking at `Mathlib.RingTheory.RootsOfUnity.Basic` I see a bunch of `0<n` and `n : ℕ+`: this is already not great, but I am afraid that adding yet another way of saying the same thing is not great. Of course the real solution is to choose once and for all what to do...

Reply:
The use of `n+1` to denote a positive natural is, I think, fairly common in Mathlib (and certainly better than using `PNat` in my opinion). The question is probably which version is easier to use. I have no strong opinion here (but hope that merging this will not be delayed because of a lack of consesus among the reviewers on this point :smile:).

Reply:
I have plans to eventually refactor roots of unity to avoid `PNat`, but this will have to wait until teaching is over.

Reply:
I think it is a good thing, we can maybe discuss it on zulip (for example, do we want to do the same in `IsCyclotomicExtension`? I think so).